### ISSUE TITLE
Creating duplicate item without a required field gives the wrong error message

### ISSUE THREAD

**Issue by AlanCoding** [ansible/tower-cli#109]
Scenario:

- Two or more hosts with a given name exists (within different inventories)
- Try to create a new host with that same name without specifying the inventory

Result:

- Error message `Error: Expected one result, got 2. Tighten your criteria.`

What it should do:

- Error message `Error: Missing required fields: inventory`

**Comment by AlanCoding** [comment]
I think this is solved by #135

**Comment by jangsutsr** [comment]
Issue is not resolved in 2.3.2 release:
```
(env) sitan-OSX:Desktop sitan$ tower-cli host create --name localhost -v --insecure
*** DETAILS: Checking for an existing record. *********************************
GET https://ec2-54-234-179-218.compute-1.amazonaws.com/api/v1/hosts/
Params: {'name': u'localhost'}

Error: Expected one result, got 2. Tighten your criteria.
```

Upgrade milestone to 2.4.0.
